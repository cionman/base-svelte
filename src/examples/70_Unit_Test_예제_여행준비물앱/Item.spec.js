import {cleanup, render} from '@testing-library/svelte';

import Item from './Item.svelte';

describe('Item', () => {
  const categoryId = 1;
  const dnd = {};
  const item = {id: 2, name: 'socks', packed: false};

  // Unmounts any components mounted in the previous test.
  afterEach(cleanup);

  test('should render', () => {
    const {getByTestId, getByText} = render(Item, {categoryId, dnd, item});
    const checkbox = document.querySelector('input[type="checkbox"]');
    expect(checkbox).not.toBeNull(); // found checkbox
    expect(getByText(item.name)); // found item name
    expect(getByTestId('delete')); // found delete button
  });

  test('should match snapshot', () => {
    const {container} = render(Item, {categoryId, dnd, item});
    expect(container).toMatchSnapshot();
  });
});
