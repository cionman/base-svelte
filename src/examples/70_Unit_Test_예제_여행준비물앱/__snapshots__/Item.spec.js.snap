// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match snapshot 1`] = `
<body>
  <div>
    <li>
      <input
        aria-label="Toggle Packed"
        type="checkbox"
      />
       
      <span
        class="packed-false"
        draggable="true"
      >
        socks
      </span>
       
      <button
        class="icon"
        data-testid="delete"
      >
        ðŸ—‘
      </button>
    </li>
  </div>
</body>
`;
